pkg/plugins/manager/loader/loader.go

```diff
@@ -44,19 +44,34 @@ func (l *Loader) Load(ctx context.Context, src plugins.PluginSource) ([]*plugins
 		return nil, err
 	}
 
-	bootstrappedPlugins, err := l.bootstrap.Bootstrap(ctx, src, discoveredPlugins)
-	if err != nil {
-		return nil, err
+	bootstrappedPlugins := []*plugins.Plugin{}
+	for _, foundBundle := range discoveredPlugins {
+		bootstrappedPlugin, err := l.bootstrap.Bootstrap(ctx, src, foundBundle)
+		if err != nil {
+			// TODO: Add error to registry
```

Comment:
We may not need it, but we could have specific a error wrapper for errors we actually want to track (and we actually just fail otherwise). 

For example
```go
const (
	StatusNoBackendBinaryFound PluginLoadingErrStatus = ""
)

type PluginLoadingErrStatus string

type StageErr struct {
    Status PluginLoadingErrStatus
}
```

```go
var stageErr StageErr

bootstrappedPlugins := []*plugins.Plugin{}
for _, foundBundle := range discoveredPlugins {
	bootstrappedPlugin, err := l.bootstrap.Bootstrap(ctx, src, foundBundle)
	if err != nil {
		if errors.As(err, &stageErr) {
			fmt.Printf("Do stuff failed with error %s, underlying error is: %s\n", err, pErr)
			registerPluginErr(ctx, foundBundle.ID, stageErr) // TODO
			continue
		}
	}
	bootstrappedPlugins = append(bootstrappedPlugins, bootstrappedPlugin...)
}
```

Reply:
as discussed offline, let's try to reuse the `SignatureError` implementation here in a generic way :+1: 

I am going to merge this to avoid as much as possible a huge PR next :crossed_fingers: